### PR TITLE
maint: implement `ansys/actions/check-vulnerabilities` in CICD

### PR DESCRIPTION
**Pull request by clatapie**
This PR is to follow PyAnsys guideline.

### CODE REVIEW

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -93,6 +93,19 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check library vulnerabilities with default mode"
+        uses: ansys/actions/check-vulnerabilities@v6.0
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          token: ${{ secrets.PYANSYS_CI_BOT_TOKEN }}
+          python-package-name: 'ansys-<product>-<library>'
+
+
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:108]: Hey.. you need to adapt things here 😄 

You will also need to add the dev version

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (clatapie) [reply .github/workflows/ci_cd.yml]: Yes, I noticed it... I pushed my commit a little bit too fast. Thanks for checking it! 😄

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -93,6 +93,33 @@ jobs:
...
+    name: "Check library vulnerabilities (development mode)"
+    if: github.ref != 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check library vulnerabilities with development mode"
+        uses: ansys/actions/check-vulnerabilities@v6.0
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          token: ${{ secrets.PYANSYS_CI_BOT_TOKEN }}
+          python-package-name: 'ansys-math-core'
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:106]:
```suggestion
          python-package-name: ${{ env.PACKAGE_NAME }}
```

File: .github/workflows/ci_cd.yml [hunk 3]
```diff
@@ -93,6 +93,33 @@ jobs:
...
+    name: "Check library vulnerabilities (default mode - only on main)"
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check library vulnerabilities with default mode"
+        uses: ansys/actions/check-vulnerabilities@v6.0
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          token: ${{ secrets.PYANSYS_CI_BOT_TOKEN }}
+          python-package-name: 'ansys-math-core'
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:120]:
```suggestion
          python-package-name: ${{ env.PACKAGE_NAME }}
```

File: .github/workflows/ci_cd.yml [hunk 4]
```diff
@@ -93,6 +93,33 @@ jobs:
...
+          dev-mode: true
+
+
+  check-vulnerabilities-main:
+    name: "Check library vulnerabilities (default mode - only on main)"
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check library vulnerabilities with default mode"
+        uses: ansys/actions/check-vulnerabilities@v6.0
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:116]:
```suggestion
        uses: ansys/actions/check-vulnerabilities@v6
```

File: .github/workflows/ci_cd.yml [hunk 5]
```diff
@@ -93,6 +93,33 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
 
+  check-vulnerabilities-dev:
+    name: "Check library vulnerabilities (development mode)"
+    if: github.ref != 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check library vulnerabilities with development mode"
+        uses: ansys/actions/check-vulnerabilities@v6.0
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:102]:
```suggestion
        uses: ansys/actions/check-vulnerabilities@v6
```

**Review by RobPasMue — APPROVED** [review]
Apart from some minor suggestions - LGTM!

**Review by RobPasMue — APPROVED** [review]
Thanks for implementing this @clatapie !

**Pull request merged**